### ISSUE TITLE
Throw a friendly error message for inconsistent data

### ISSUE THREAD

**Issue by zhichao-li** [intel-analytics/BigDL#2467]
We should throw a friendly error message when the inputs of `Model` is 3 items buy got 2 items for the training data. 

java.util.NoSuchElementException: key not found: 3
	at scala.collection.MapLike$class.default(MapLike.scala:228)
	at scala.collection.AbstractMap.default(Map.scala:58)
	at scala.collection.mutable.HashMap.apply(HashMap.scala:64)
	at com.intel.analytics.bigdl.utils.Table.apply(Table.scala:91)
	at com.intel.analytics.bigdl.nn.Graph.getInput(Graph.scala:298)
	at com.intel.analytics.bigdl.nn.Graph.findInput(Graph.scala:306)
	at com.intel.analytics.bigdl.nn.StaticGraph.updateOutput(StaticGraph.scala:57)
	at com.intel.analytics.bigdl.nn.keras.KerasLayer.updateOutput(KerasLayer.scala:187)
	at com.intel.analytics.bigdl.nn.abstractnn.AbstractModule.forward(AbstractModule.scala:243)
	at com.intel.analytics.bigdl.optim.DistriOptimizer$$anonfun$5$$anonfun$6$$anonfun$apply$2.apply$mcI$sp(DistriOptimizer.scala:250)
	at com.intel.analytics.bigdl.optim.DistriOptimizer$$anonfun$5$$anonfun$6$$anonfun$apply$2.apply(DistriOptimizer.scala:243)
	at com.intel.analytics.bigdl.optim.DistriOptimizer$$anonfun$5$$anonfun$6$$anonfun$apply$2.apply(DistriOptimizer.scala:243)
	at com.intel.analytics.bigdl.utils.ThreadPool$$anonfun$1$$anon$4.call(ThreadPool.scala:112)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at com.intel.analytics.bigdl.optim.DistriOptimizer$$anonfun$5$$anonfun$8.apply(DistriOptimizer.scala:262)
	at com.intel.analytics.bigdl.optim.DistriOptimizer$$anonfun$5$$anonfun$8.apply(DistriOptimizer.scala:262)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:47)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
	at scala.collection.AbstractTraversable.map(Traversable.scala:105)
	at com.intel.analytics.bigdl.optim.DistriOptimizer$$anonfun$5.apply(DistriOptimizer.scala:262)
	at com.intel.analytics.bigdl.optim.DistriOptimizer$$anonfun$5.apply(DistriOptimizer.scala:202)
	at org.apache.spark.rdd.ZippedPartitionsRDD2.compute(ZippedPartitionsRDD.scala:88)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:306)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:270)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:66)
	at org.apache.spark.scheduler.Task.run(Task.scala:89)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:227)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	... 1 more
Caused by: Layer info: Model[2bc0e421]
java.util.NoSuchElementException: key not found: 3
	at scala.collection.MapLike$class.default(MapLike.scala:228)
	at scala.collection.AbstractMap.default(Map.scala:58)
	at scala.collection.mutable.HashMap.apply(HashMap.scala:64)
	at com.intel.analytics.bigdl.utils.Table.apply(Table.scala:91)
	at com.intel.analytics.bigdl.nn.Graph.getInput(Graph.scala:298)
	at com.intel.analytics.bigdl.nn.Graph.findInput(Graph.scala:306)
	at com.intel.analytics.bigdl.nn.StaticGraph.updateOutput(StaticGraph.scala:57)
	at com.intel.analytics.bigdl.nn.keras.KerasLayer.updateOutput(KerasLayer.scala:187)
	at com.intel.analytics.bigdl.nn.abstractnn.AbstractModule.forward(AbstractModule.scala:243)
	at com.intel.analytics.bigdl.optim.DistriOptimizer$$anonfun$5$$anonfun$6$$anonfun$apply$2.apply$mcI$sp(DistriOptimizer.scala:250)
	at com.intel.analytics.bigdl.optim.DistriOptimizer$$anonfun$5$$anonfun$6$$anonfun$apply$2.apply(DistriOptimizer.scala:243)
	at com.intel.analytics.bigdl.optim.DistriOptimizer$$anonfun$5$$anonfun$6$$anonfun$apply$2.apply(DistriOptimizer.scala:243)
	at com.intel.analytics.bigdl.utils.ThreadPool$$anonfun$1$$anon$4.call(ThreadPool.scala:112)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

	at com.intel.analytics.bigdl.nn.abstractnn.AbstractModule.forward(AbstractModule.scala:249)
	at com.intel.analytics.bigdl.optim.DistriOptimizer$$anonfun$5$$anonfun$6$$anonfun$apply$2.apply$mcI$sp(DistriOptimizer.scala:250)
	at com.intel.analytics.bigdl.optim.DistriOptimizer$$anonfun$5$$anonfun$6$$anonfun$apply$2.apply(DistriOptimizer.scala:243)
	at com.intel.analytics.bigdl.optim.DistriOptimizer$$anonfun$5$$anonfun$6$$anonfun$apply$2.apply(DistriOptimizer.scala:243)
	at com.intel.analytics.bigdl.utils.ThreadPool$$anonfun$1$$anon$4.call(ThreadPool.scala:112)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 3 more
